### DISCUSSION TITLE
Компиляция под mipsel 24kc

### DISCUSSION THREAD

**Discussion by MerQury14092** [hufrea/byedpi#70] [General]
Есть роутер Xiaomi Mi Router 4c с прошивкой OpenWrt
`opkg print-architecture` выдает
```
arch all 1
arch noarch 1
arch mipsel_24kc 10
```

Пробовал скомпилировать командой 
`mips-linux-gnu-gcc -march=24kc -O2 packets.c main.c conev.c proxy.c desync.c mpool.c extend.c -I . -o ciadpi`

выдает файл, который при попытке запуска на роутере выдает ошибку
`/tmp/ciadpi: line 1: syntax error: unexpected "("`

Пробовал так же архитектуры
`24kec`, `mips1`, `mips32r2`, выдает такие же ошибки

на роутере не могу скомпилировать, ведь gcc вмесит 40 МиБ, на роутере столько места нет

**Comment by hufrea** [comment]
Полагаю вы используете компилятор для  mips, когда нужно под mipsel.
mips-linux-gnu-gcc -> gcc-mipsel-linux
Тут есть уже готовые сборки: https://github.com/spvkgn/ByeDPI-OpenWrt/releases

**Comment by MartynIden** [comment]
а как лучше запускать это дело под Openwrt? поставил ipk под mipsel 24kc, что дальше делать с ним?)

**Comment by spvkgn** [comment]
@MartynIden править конфиг под свои нужды - `/etc/config/byedpi` и перезапустить `service byedpi restart`
